### PR TITLE
Run the code formatter on Mix.ProjectStack

### PR DESCRIPTION
**Pull request by PChambino**

### CODE REVIEW

File: lib/mix/lib/mix/project_stack.ex [hunk 1]
```diff
@@ -2,188 +2,201 @@ defmodule Mix.ProjectStack do
   @moduledoc false
 
   use Agent
-  @timeout 30_000
+  @timeout 30000
```
Comment (PChambino) [inline lib/mix/lib/mix/project_stack.ex]: I think it would be nice if the formatter didn't remove underscores from numbers.

Comment (ivan) [reply lib/mix/lib/mix/project_stack.ex]: It formats with underscores for numbers with >= 6 digits.

File: lib/mix/lib/mix/project_stack.ex [hunk 2]
```diff
@@ -2,188 +2,201 @@ defmodule Mix.ProjectStack do
...
   defp find_project_named(name, stack) do
-    name && Enum.find_value(stack, fn
-      %{name: n, file: file} when n === name -> file
-      %{} -> nil
-    end)
+    name &&
+      Enum.find_value(stack, fn
+        %{name: n, file: file} when n === name -> file
+        %{} -> nil
+      end)
```
Comment (whatyouhide) [inline lib/mix/lib/mix/project_stack.ex]: Can you please refactor this to an `if`?

```elixir
if name do
  name
else
  Enum.find_value(stack, fn
    ...
  end)
end
```

**Comment by whatyouhide** [comment]
Thanks @PChambino! 💟

File: lib/mix/lib/mix/project_stack.ex (hunk 2 above)
Comment (PChambino) [reply lib/mix/lib/mix/project_stack.ex]: The correct refactoring using an `if` is
```elixir
if name do
  Enum.find_value(...)
end
```
Would matching on the `name` argument be preferable?
```elixir
  defp find_project_named(nil, _stack), do: nil

  defp find_project_named(name, stack) do
    Enum.find_value(...)
  end
```

**Pull request merged**